onnx/defs/nn/defs.cc

```diff
@@ -1363,6 +1363,93 @@ output_shape can also be explicitly specified in which case pads values are auto
 
 ONNX_OPERATOR_SET_SCHEMA(ConvTranspose, 11, OpSchema().FillUsing(ConvTransposeOpSchemaGenerator("a filter")));
 
+static const char* DeformConv_ver19_doc = R"DOC(
+Performs Deformable Convolution as described in https://arxiv.org/abs/1703.06211 and https://arxiv.org/abs/1811.11168.
+)DOC";
+
+ONNX_OPERATOR_SET_SCHEMA(
+    DeformConv,
+    19,
+    OpSchema()
+        .SetDoc(DeformConv_ver19_doc)
+        .Input(
+            0,
+            "X",
+            "Input data tensor. For 2D image data, it has shape (N, iC, H, W) where N is the batch size, "
+            "iC is the number of input channels, and H and W are the height and width. "
+            "In general, the shape is (N, iC, D1, D2, ... , Dn) for n-dimensional data, where "
```

Comment:
Is the general n-dimensional version used? Or, would it be enough to have just the 2D version or 2D/3D version?

Reply:
All applications in the original paper are 2D. I see `Conv` includes support for arbitrary dimensions. Should we be consistent?

Reply:
While generalizing the spec of an op for higher dimensions is relatively easy, it can be non-trivial to generalize an implementation (I am commenting generally based on experience with other ops, I don't know enough about this specific op). I often see specialized implementations for 2 or 3 D, with a separate generic implementation for the generic N-D case.

In such cases, it seems to me that it is justifiable to define the 2/3 D version if that is the only one used in practice.

Reply:
@gramalingam I added a sentence to the op description and the input description saying the common use cases are 2D and 3D, after consulting internal stakeholders. Let me know if you're ok with that. Thanks!